build-tools/packages/build-cli/src/commands/check/policy.ts

```diff
@@ -0,0 +1,243 @@
+                        const resolver = handler.resolver;
+                        if (flags.resolve && resolver) {
+                            output += `${newline}attempting to resolve: ${file}`;
+                            const resolveResult = runWithPerf(handler.name, "resolve", () =>
+                                resolver(file, pathToGitRoot),
+                            );
+
+                            if (resolveResult.message) {
+                                output += newline + resolveResult.message;
+                            }
+
+                            if (!resolveResult.resolved) {
+                                this.exit(1);
+                            }
+                        } else {
+                            this.exit(1);
+                        }
+
+                        this.log(output);
+                    }
+                });
+        };
+
+        const handleLine = (line: string) => {
+            const filePath = path.join(pathToGitRoot, line).trim().replace(/\\/g, "/");
+
+            if (pathRegex.test(line) && fs.existsSync(filePath)) {
+                count++;
+                if (exclusions.every((value) => !value.test(line))) {
+                    try {
+                        routeToHandlers(filePath);
+                    } finally {
+                        runPolicyCheck();
+                        logStats();
+                    }
+
+                    processed++;
+                } else {
+                    this.log(`Excluded: ${line}`);
+                }
+            }
+        };
+
+        const runPolicyCheck = () => {
+            for (const h of handlers) {
+                const final = h.final;
+                if (final) {
+                    const result = runWithPerf(h.name, "final", () =>
+                        final(pathToGitRoot, flags.resolve),
+                    );
+                    if (result?.error) {
+                        this.exit(1);
+                        this.log(result.error);
+                    }
+                }
+            }
+        };
+
+        const logStats = () => {
+            this.log(
+                `Statistics: ${processed} processed, ${count - processed} excluded, ${count} total`,
+            );
+            for (const [action, handlerPerf] of handlerActionPerf.entries()) {
+                this.log(`Performance for "${action}":`);
+                for (const [handler, dur] of handlerPerf.entries()) {
+                    this.log(`\t${handler}: ${dur / 1000}:`);
+                }
+            }
+        };
+
+        if (flags.resolve) {
+            this.log("Resolving errors if possible.");
+        }
+
+        if (flags.handler !== undefined) {
+            this.log(`Filtering handlers by regex: ${handlerRegex}`);
+        }
+
+        if (flags.path !== undefined) {
+            this.log(`Filtering file paths by regex: ${pathRegex}`);
+        }
```

Comment:
I think the `run` method will be easier to understand if you move most of the functions aboce this to be defined either on the class or outside the class. Then the run function body is smaller and easier to read and understand end-to-end.